### DISCUSSION TITLE
Help with MHI2_US_AUG11_S0533

### DISCUSSION THREAD

**Discussion by MrGonk** [Mr-MIBonk/M.I.B._More-Incredible-Bash#82] [Q&A] [answered]
Hi: I'm sorry to the extent that this is a stupid question/could've found this somewhere in a FAQ, but I'm afraid I'm such a novice, I don't even know how to begin to resolve this question.

I have a 16 Audi A7 with MHI2_US_AUG11_S0533. I tried M.I.B. v. 1.2, both automatic and red menu manual install approaches. Regardless of how I executed, the update ran exactly as described in the Wiki and accompanying video explainers, except that at the end, in the GEM, there was no M.I.B. option. It just wasn't there - it was exactly the same as before. 

I see there is no patch that appears to correspond to MHI2_US_AUG11_S0533, and I did see one thread where someone was asking about having trouble converting to EU firmware from MHI2_US_AUG11_S0533, but I honestly couldn't tell how on-point that was. I just want to install the tool and run the man feature patches (e.g., AA/ACP activation, VIM activation, map updates, etc.). I just don't know where to go from here. Do I need to find a newer US FW version I think I can flash to? Or do I need to try to use the old direct D-Link method or something? 

Any point in the right direction would be much appreciated. Thank you.

**Comment by harman-f** [comment]
M.I.B 1.2 is very old!

Also your FW version is most likely not supported by M.I.B.

Try with latest M.I.B version.
You most likely have to update your FW 1st.

**Comment by MrGonk** [comment]
Sorry - to clarify, I used v 3.4.1 ("Backup Edition") of M.I.B., which appeared to attempt to install M.I.B. Installer v. 1.2. I unpacked the full batch of patches into the patch directory on the SD card, as well. 

My existing FW is not listed on the compatibility table, so, as you say, it looks like it's too old. If anyone has any insight into how best I can update my FW to something compatible that a 2016 C7.5 with 4G MIB can use (I assume it can't necessarily use just any U.S. firmware), I'd be grateful. I dug around here and didn't see anything (e.g., in the Wiki) to indicate how I can update the FW itself. 

Many thanks, again.

Edit: Actually, check that. After poking around more I found the FW repository with all the train compatibility info right there in the file detail. I'm going to try to update my US S0533 to MHI2_US_AU57x_K0539_MU1602 and see if I can install M.I.B. on that.... Thanks again.

**Reply by harman-f** [reply]
Check on your SD card if you find a backup in `/backup/`.
If so, zip it and share.

**Reply by MrGonk** [reply]
Nothing in there. Seems to me like you need to be able to get into the m.i.b. selection in GEM to initiate a backup, and I can't get the m.i.b. menu into GEM. 

I'm also discovering that if the problem is my FW version, it looks like I'm going to have a hell of a time updating this FW to anything. Apparently the "S" train goes essentially nowhere, and the only way to attempt an update would be to edit the meta file, which I am still trying to find information on how to do.... 

Thanks, guys.

**Reply by harman-f** [reply]
> Nothing in there. Seems to me like you need to be able to get into the m.i.b. selection in GEM to initiate a backup, and I can't get the m.i.b. menu into GEM.
> 
Very old FW versions do not contain certain binaries. During M.I.B install a basic backup is generated, but if even that is not there, a lot a stuff must be missing.

> I'm also discovering that if the problem is my FW version, it looks like I'm going to have a hell of a time updating this FW to anything. Apparently the "S" train goes essentially nowhere, and the only way to attempt an update would be to edit the meta file, which I am still trying to find information on how to do....
> 
> Thanks, guys.

You mean MHI2_US_AU57x_K0539_MU1062 (A6-A7)?
Is tehre a reason not to go to MHI2_US_AU57x_K3341_MU1426?

I can help you with that modified metainfo, just tell me, which FW you want to install

**Reply by harman-f** [reply]
However, I do not have experience with upgrading on the US train.
Have you heard about any special steps while doing this?

Does MHI2_US_AU57x_K0539_MU1062 (A6-A7) already support CarPlay?
Sometimes it is better to stay within a train - 05XX - and do not upgrade to 33XX

**Reply by MrGonk** [reply]
K0539 was what I'd been trying to update to. It looks like it's compatible with M.I.B. but doesn't come with CarPlay or Android Auto already enabled. My understanding is that K3341 is the absolute latest firmware made for the C7.5 4G MIB2 platform and, I suspect, would come already with ACP and AA, along with whatever else. 

I'm not sure what the risks are exactly of going outside of a train, but there seems to be a lot of confidence in K3341 among the A6/A7 crowd. 

I got up on the Telegram and a kind-hearted user actually modded the metainfo2 file for me. I've attached it and the original from K33341 in case anyone has any comments on it or might find it useful. The standard-named one is the modded version; the one with the funky name is the original. 

[_metainfo2(original).txt](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/files/10326678/_metainfo2.original.txt)
[metainfo2.txt](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/files/10326679/metainfo2.txt)

**Reply by harman-f** [reply]
Did you already update and patch?
CarPlay working?

**Reply by MrGonk** [reply]
I haven't yet. I'm thinking I'll do it this afternoon. Does the modded file look rigtht? Kinda nervous about that, as I really have no way of judging and definitely don't want to brick my car (though I understand that is highly unlikely).

**Reply by willblk973** [reply]
I have used that file to update mine and it worked, but at the last module I had to change the file back to the original for the install the recognize and continue. patched car-play with the MIB. Thanks for providing this file!

**Reply by MrGonk** [reply]
> I have used that file to update mine and it worked, but at **the last module I had to change the file back to the original for the install the recognize and continue**. patched car-play with the MIB. Thanks for providing this file!

-- Can you elaborate on exactly what you had to do? I'm not sure what exactly you mean by the part I bolded above. Just want to be sure I have a clear idea if some part of the process might have to be tweaked. Thanks!

**Reply by willblk973** [reply]
After starting the install using the updated metafile and the installation getting to part 10/12 of the update after a reboot the MMI stated it could not read the metainfo2 file and gave the option to "try again" did that and it did not work. At that point I ejected the SD card went back to my computer and deleted the "fixed" version of the file with the original metainfo2 file. I then put the SD back into the car and hit the "TRY AGAIN" prompt at that point the install continued to the end with no other issue. and is now MHI2_US_AU57x_K3341_MU1426 from MHI2_US_AUG11_S0533. working perfectly. Additionally updated the Maps and POIs.

**Reply by MrGonk** [reply]
I've hit a snag. So, got the new FW installed, all went smoothly. Then I went to install MIB 3.4.1, which also appeared to install perfectly with no errors. However, now I can't get into the GEM, at all. Even 5+ minutes after a reboot. Note that I'd had access to it before (enabled it with OBDEleven). 

... Any ideas?? Super grateful as always.
![0101231731_HDR](https://user-images.githubusercontent.com/121593286/210186811-cb9d2dd4-ae30-4e39-b330-6b9d7eede93b.jpg)
![0101231707_HDR](https://user-images.githubusercontent.com/121593286/210186814-ddc87a07-ead3-4001-a97c-6a894928c15f.jpg)

**Reply by harman-f** [reply]
do you see files on your SD card in /backup/?

ENtering GEM sometimes can be troublesome. SUre you got the button combination right?

**Comment by MrGonk** [comment]
Thanks, man. Really appreciate the color - very helpful.

**Reply by willblk973** [reply]
you're welcome. After that I suggest reinstalling the MIB along with the patch file for MHI2_US_AU57x_K3341_MU1426. once that is installed go to the aio (ALL_IN _ONE) and run that to get CarPlay and AA also the FECs needed for the new MAPS

**Reply by MrGonk** [reply]
Luckily, just got past the post-module 10 reboot without the error you described. Third reboot underway.
![0101231658_HDR](https://user-images.githubusercontent.com/121593286/210185819-1ced8e83-6c2f-4f9e-9db0-4fee3d689639.jpg)

**Reply by harman-f** [reply]
The error befor is strange. 
@willblk973 I would redo the FW update, could be that some module was not updated. ALso changing metainfo2.txt mid update is not the best idea ;)

**Reply by Zaybar69** [reply]
Hi . Can you please help me with this issue? I want to install the 1426 fw but I don't know how . I downloaded and extract it with winrar and I have now 24 files. I put them on the directory of the SD card or I have to put them in another folder. Because when I try to update with red menu , it says that there is no files on SD . Can you share a picture how it should be the files on the SD ? Thx

**Reply by AudiSatoshi** [reply]
> Hi . Can you please help me with this issue? I want to install the 1426 fw but I don't know how . I downloaded and extract it with winrar and I have now 24 files. I put them on the directory of the SD card or I have to put them in another folder. Because when I try to update with red menu , it says that there is no files on SD . Can you share a picture how it should be the files on the SD ? Thx 

First you need to make sure the SD card is in fat32 not exfat or anything else. Go to my computer and right click properties on the drive to see this information. If you need to format to get to fat32 there is a formatter integrated into windows just right click on the drive and click format and select fat32. If you don't see anything like this for your setup there is a fat32 formatter in the MIB one website. Look in the github for the link or search online. (You can't do this on mac unless you have some sort of VM to windows or modify it) 
Secondly you should use a full sized SD card not a micro sd with an SD card adapter.(because they are usually formatted in exfat by standard and less reliable than full sized SDs) As far as the files, you don't have to do anything except extract/unzip the zipped/rar files directly into the SDs drive (thats what I did) You wont be able to do that for the MIB  zipped/rar because there is another subfolder but all you do for that is extract to desktop (or anywhere on your computers drive) and open the subfolder and then you are good to copy paste the files over to the SD card's drive directory.

**Comment by MrGonk** [comment]
I'll definitely load up M.I.B. again after. Is there an AIO patch for US? I've only seen EU, but I may be confused.

**Comment by MrGonk** [comment]
Installing the new firmware now. On package 4/12 right now; so far everything looking good, knock (luxurious open pore) wood. Thanks for all the help here.

**Comment by MrGonk** [comment]
OK, I have to be doing something fundamentally wrong here, or have missed some kind of important step. After trying the installation process again, I think M.I.B. is simply not installing at all, and the process isn't even generating any logs or anything. I have no log to share, because there's nothing in there. I could really use some help on what is going wrong. 

To recap: 
- The vehicle is a 2016 Audi A7 TDI Prestige with the emissions fix sold as new in 2018. It had FW version MHI2_US_AUG11_S0533.
- When I first tried to install M.I.B. with that original firmware, I *was* able to access the GEM, no "m.i.b." menu item came up in my GEM. No logs were generated. I think the only reason I was able to get into the GEM at that time was because I'd activated it using OBD11 months ago.
- I learned that my original firmware version was both old and incompatible with M.I.B. Thanks to the help of folks here and on Telegram, I was able to install MHI2_US_AU57x_K3341_MU1426, which is the latest version compatible with my unit, using an edited metainfo file. The firmware update went perfectly, and now I'm running that firmware with no issues whatsoever. 
- MHI2_US_AU57x_K3341_MU1426 is M.I.B.-compatible according to the table.
- I ran the process described in the Wiki exactly as described to install M.I.B. v. 3.4.1 with the updated firmware in place. Everything proceeded and presented as described in the Wiki and the Audi-specific YouTube walkthrough, up to the point where you're supposed to open the GEM.
- After running the install, which showed "Launcher" as "OK" and no errors (everything else was "[-]"), I was unable to access the GEM at all. I suspect this is because updating from S0533 to K3341 somehow undid the GEM access I'd activated with OBD11 (however, it doesn't appear to have affected certain other changes I made with OBD11, such as unlocking the "efficiency" drive mode.
- I know installing M.I.B. is supposed to activate GEM access, and I still can't get into GEM.
- I've used the autorun approach (removing the underscore from the "_swdlautorun" file) and the manual REM method, and I've noticed that when I've used the autorun approach, once the M.I.B. install is done, when the unit finally reboots, I see I have audio, nav and normal controls, and as soon as everything is initialized, M.I.B. install starts again. It'll run over and over if you leave the SD card in. The guides certainly seem to suggest it shouldn't attempt to install again after M.I.B. has installed, so that seems further evidence that it's just not installing. 
- I've tried it with all of the patches in the patch menu and with just the K3341 patch; I've tried to install probably at least six times now. I've re-formatted and re-compiled the SD card multiple times in case I'd made a mistake putting the files together.
- No interruptions ever occurred in any of the installations, and none have ever generated a log or backup.
- I can provide pictures of the various steps of the FW update and M.I.B. install.

Help would really be appreciated. This is driving me nuts.

**Reply by harman-f** [reply]
Check you SD card for backup and log file.

If you find them, share them here.

**Reply by MrGonk** [reply]
There are none. Just the original folder structure and empty placeholder files (e.g., "logs go here").

**Reply by harman-f** [reply]
This is very unusual.
After upgrading to K3341 there is no reason, that M.I.B should not work.

Get a fresh SD card and try again!

**Comment by MrGonk** [comment, marked as answer]
Well, this is embarrassing. The good news is I got into the GEM, and everything works as advertised. The bad news is I'm a huge idiot. 

The card was write-protected, and by dumbest possible means. I was using a micro-to-standard SD adapter, and sure enough, the slider was in the "locked" position *every time* I tried to install MIB. My computer takes the micro directly, which obviously doesn't have the slider lock, which is why I was able to write the files to it but not install once it was in the car. Moved the slider to unlocked position, everything ran perfectly. 

Thanks to all the many people who reached out with advice and help. Your knowledge and patience are really incredibly impressive. I really appreciate it and apologize for needing this much help with something this bone-headed. Thanks again!!

**Comment by defray23** [comment]
Could somebody kindly point me in the right direction on which firmware I need to upgrade to and how to install it?

I also have an Audi A7 2016 with he same base firmware as OP (MHI2_US_AUG11_S0533) and tried installing MIB 3.4.1 which ran without issues and showed LAUNCHER OK but I cannot enter GEM and presumably MIB can't run on this old firmware.

On mibsolution.one I see newer firmware like MHI2_US_AU57x_K0539_MU1062 (A6-A7) but the supported train doesn't show my version specifically, a very close one though (MHI2_US_AUG11_P0533).
Is this the right FW to upgrade to in order to be able to run MIB after?
EDIT: Actually found MU1426 that worked for OP via google search on mibsolution and downloading that now as that seems to work and is confirmed supported by MIB.

I can upgrade with the same red menu option I guess? But want to avoid issues with missing edits in the metainfo file.
Anybody could kindly point me in the right direction?
Sorry didn't see an option to PM MrGonk specifically via GitHub.

UPDATE: 
Nevermind, I ran the MU1426 update with the custom metainfo file in this chat and it ran without issues (only SDARS and LTE updates failed but seemed to have no effect). I then ran the MIB install, got the green menu, did all the little tweaks and CarPlay activation etc., everything works flawlessly on first glance.
Now I just need to update my maps and I'm all set, kudos to all the contributors to this great piece of software!

**Reply by harman-f** [reply]
Check this post:
https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/discussions/82#discussioncomment-4558609

And the discussion between @MrGonk and @willblk973

**Comment by Zaybar69** [comment]
Bi , and thank you Very much for your answers. I did all that , but no result . This way should be the files inside the SD ? 

Yahoo Mail: cerca, organizza, prendi il controllo della tua casella di posta 
 
  Il gio, 12 dic, 2024 alle 15:47, ***@***.***> ha scritto:   



Hi . Can you please help me with this issue? I want to install the 1426 fw but I don't know how . I downloaded and extract it with winrar and I have now 24 files. I put them on the directory of the SD card or I have to put them in another folder. Because when I try to update with red menu , it says that there is no files on SD . Can you share a picture how it should be the files on the SD ? Thx


First you need to make sure the SD card is in fat32 not exfat or anything else. Go to my computer and right click properties on the drive to see this information. If you need to format to get to fat32 there is a formatter integrated into windows just right click on the drive and click format and select fat32. If you don't see anything like this for your setup there is a fat32 formatter in the MIB one website. Look in the github for the link or search online. (You can't do this on mac unless you have some sort of VM to windows or modify it)
Secondly you should use a full sized SD card not a micro sd with an SD card adapter.(because they are usually formatted in exfat by standard and less reliable than full sized SDs) As far as the files, you don't have to do anything except extract/unzip the zipped/rar files directly into the SDs drive (thats what I did) You wont be able to do that for the MIB zipped/rar because there is another subfolder but all you do for that is extract to desktop (or anywhere on your computers drive) and open the subfolder and then you are good to copy paste the files over to the SD card's drive directory.

**Comment by Zaybar69** [comment]
Like this ? This is what I have after I extracted the folder 1426 . Can you send me a link pls ? Maybe this one is corrupt 

Yahoo Mail: cerca, organizza, prendi il controllo della tua casella di posta 
 
  Il gio, 12 dic, 2024 alle 15:47, ***@***.***> ha scritto:   



Hi . Can you please help me with this issue? I want to install the 1426 fw but I don't know how . I downloaded and extract it with winrar and I have now 24 files. I put them on the directory of the SD card or I have to put them in another folder. Because when I try to update with red menu , it says that there is no files on SD . Can you share a picture how it should be the files on the SD ? Thx


First you need to make sure the SD card is in fat32 not exfat or anything else. Go to my computer and right click properties on the drive to see this information. If you need to format to get to fat32 there is a formatter integrated into windows just right click on the drive and click format and select fat32. If you don't see anything like this for your setup there is a fat32 formatter in the MIB one website. Look in the github for the link or search online. (You can't do this on mac unless you have some sort of VM to windows or modify it)
Secondly you should use a full sized SD card not a micro sd with an SD card adapter.(because they are usually formatted in exfat by standard and less reliable than full sized SDs) As far as the files, you don't have to do anything except extract/unzip the zipped/rar files directly into the SDs drive (thats what I did) You wont be able to do that for the MIB zipped/rar because there is another subfolder but all you do for that is extract to desktop (or anywhere on your computers drive) and open the subfolder and then you are good to copy paste the files over to the SD card's drive directory.

**Comment by Zaybar69** [comment]
Can you please help me ? I can offer money if you want . 

Yahoo Mail: cerca, organizza, prendi il controllo della tua casella di posta 
 
  Il gio, 12 dic, 2024 alle 15:47, ***@***.***> ha scritto:   



Hi . Can you please help me with this issue? I want to install the 1426 fw but I don't know how . I downloaded and extract it with winrar and I have now 24 files. I put them on the directory of the SD card or I have to put them in another folder. Because when I try to update with red menu , it says that there is no files on SD . Can you share a picture how it should be the files on the SD ? Thx


First you need to make sure the SD card is in fat32 not exfat or anything else. Go to my computer and right click properties on the drive to see this information. If you need to format to get to fat32 there is a formatter integrated into windows just right click on the drive and click format and select fat32. If you don't see anything like this for your setup there is a fat32 formatter in the MIB one website. Look in the github for the link or search online. (You can't do this on mac unless you have some sort of VM to windows or modify it)
Secondly you should use a full sized SD card not a micro sd with an SD card adapter.(because they are usually formatted in exfat by standard and less reliable than full sized SDs) As far as the files, you don't have to do anything except extract/unzip the zipped/rar files directly into the SDs drive (thats what I did) You wont be able to do that for the MIB zipped/rar because there is another subfolder but all you do for that is extract to desktop (or anywhere on your computers drive) and open the subfolder and then you are good to copy paste the files over to the SD card's drive directory.

**Reply by cudddy** [reply]
I have version MHI2_US_AU57x_S0536.  Do I need to upgrade to K3341 to make this work?  How do I upgrade to this version. Very much a novice but can follow instructions.  I drive a ton and my 16 A7 would be so much nicer with carplay

**Comment by Zaybar69** [comment]
Can you please help me. I can pay if you want. The k3341 it's not the same inside as the k3663 . I try to install k3341 and makes me skip files . You can see how it's inside k3663 and how it's k3441.  I have tried everything. Even to change metainfo 






Yahoo Mail: cerca, organizza, prendi il controllo della tua casella di posta 
 
  Il gio, 12 dic, 2024 alle 15:47, ***@***.***> ha scritto:   



Hi . Can you please help me with this issue? I want to install the 1426 fw but I don't know how . I downloaded and extract it with winrar and I have now 24 files. I put them on the directory of the SD card or I have to put them in another folder. Because when I try to update with red menu , it says that there is no files on SD . Can you share a picture how it should be the files on the SD ? Thx


First you need to make sure the SD card is in fat32 not exfat or anything else. Go to my computer and right click properties on the drive to see this information. If you need to format to get to fat32 there is a formatter integrated into windows just right click on the drive and click format and select fat32. If you don't see anything like this for your setup there is a fat32 formatter in the MIB one website. Look in the github for the link or search online. (You can't do this on mac unless you have some sort of VM to windows or modify it)
Secondly you should use a full sized SD card not a micro sd with an SD card adapter.(because they are usually formatted in exfat by standard and less reliable than full sized SDs) As far as the files, you don't have to do anything except extract/unzip the zipped/rar files directly into the SDs drive (thats what I did) You wont be able to do that for the MIB zipped/rar because there is another subfolder but all you do for that is extract to desktop (or anywhere on your computers drive) and open the subfolder and then you are good to copy paste the files over to the SD card's drive directory.

**Reply by AudiSatoshi** [reply]
This thread is only about K3341 because its the one compatible for 2016 Audi. I would try to install that one first and then install MIB. Then reason the other is most likely not compatible with your vehicle. Do some research and find out what you need for your car. All the information is here.

**Comment by Zaybar69** [comment]
But I have the exact firmware you had,33. I saw that you used another metainfo. I try to use it also , but no luck . 

Yahoo Mail: cerca, organizza, prendi il controllo della tua casella di posta 
 
  Il sab, 14 dic, 2024 alle 23:08, ***@***.***> ha scritto:   


This thread is only about K3341 because its the one compatible for 2016 Audi. I would try to install that one first and then install MIB. Then reason the other is most likely not compatible with your vehicle. Do some research and find out what you need for your car. All the information is here.

**Comment by vadik1380** [comment]
Hi, I have the same car with the same firmware version, can you help me please? Text me on telegram @vadd1k

**Comment by mackeydesigns** [comment]
So I just did all of this (thanks for the info btw). I was able to successfully update my firmware (red menu) and was able to install MIB (green menu) and run the backups and activate Carplay, however..... I don't get that option when connecting a device in my car. I've tried my iPhone 15 and my sons iPhone 13, i was able to get his to show up under USB device, but Carplay still wouldn't appear.

Any chance you, or someone else in this tread has had a similar issue? I've verified Carplay and AA are both activated in the green menu. Not sure what else to try at this point other than more USB cords.

**Comment by justelj1375** [comment]
I tried to update my 2016 Audi A6 (C7.5) MMI from: MHI2_US_AUG11_S0533 MU 0143 --> MHI2_US_AUG57x_K3341_1 MU 1426
Now my MMI is stuck in Emergency Assist Utility reboot loop trying to fix RCC efs and I keep getting file size doesnt match metainfo tag??
Does anyone have the old firmware?
Does anyone have an idea what to do to fix this?
Im stumped and sick to my stomach.....

**Reply by Xitro01** [reply]
> I tried to update my 2016 Audi A6 (C7.5) MMI from: MHI2_US_AUG11_S0533 MU 0143 --> MHI2_US_AUG57x_K3341_1 MU 1426 Now my MMI is stuck in Emergency Assist Utility reboot loop trying to fix RCC efs and I keep getting file size doesnt match metainfo tag?? Does anyone have the old firmware? Does anyone have an idea what to do to fix this? Im stumped and sick to my stomach.....

I'm in somewhat the same situation, also after a firmware update that went wrong while writing RCC. How did you eventually get this fixed?

**Reply by justelj1375** [reply]
I downloaded the firmware again from a different sire and redid the SD cards.
I had unplugged the 5A fuse from the truck to shut off the MMI because it was stuck in a loop.
I gave it a day.. then put the redone SD card back it.. reinserted the 5A fuse.. and it took off from there.
It managed to "fix itself" and get past the checksum error it was stuck on and then finished updating the firmware.
Everything works as it should.

Maybe was a bad file in the beginning? Either way.. it was scary as I thought I ruined my MMI.
